ams/ams-api/src/main/java/com/netease/arctic/ams/api/client/ArcticThriftUrl.java

```diff
@@ -74,17 +80,43 @@ public static ArcticThriftUrl parse(String url) {
           zkServerAddress = m.group(1);
           cluster = m.group(2);
         }
-        ZookeeperService zkService = ZookeeperService.getInstance(zkServerAddress);
-        AmsServerInfo serverInfo = null;
-        try {
-          serverInfo = JSONObject.parseObject(
-              zkService.getData(AmsHAProperties.getMasterPath(cluster)),
-              AmsServerInfo.class);
-        } catch (Exception e) {
-          throw new RuntimeException("get master server info from zookeeper error", e);
+        int maxRetries = 3;
+        int retryCount = 0;
+        boolean success = false;
+
+        while (!success && retryCount < maxRetries) {
+          try {
+            ZookeeperService zkService = ZookeeperService.getInstance(zkServerAddress);
+            AmsServerInfo serverInfo = null;
```

Comment:
i think we can merge the two sentences to one!
> AmsServerInfo serverInfo = null;
            serverInfo = JSONObject.parseObject(
                zkService.getData(AmsHAProperties.getMasterPath(cluster)),
                AmsServerInfo.class);